### PR TITLE
The emag telefrag

### LINKED ISSUE #29102: [Feature Request] Telefragging
[featurerequest]
Making it a feature request instead of coding it due to probable controversy
Telefragging consists of gibbing or delimbing anyone standing on top of tracking beacons or telescience coordinates when someone is teleported onto them, like in quake or TF2 or something.

### PR DESCRIPTION
**Pull request by ghost**
[content][tested]
oooot
Closes #29102.
Should be a somewhat fair way to do it.
Plays this on success: https://files.catbox.moe/g44h2v.ogg

:cl:
 * rscadd: Emagged tracking beacons will no longer use collision safety protocols on teleports to them, resulting in the gibbing of anything on the receiving end smaller in size than the teleported mob. Does not happen if teleported mob is standing and receiving one is lying, or vice versa.

### CODE REVIEW

**Comment by pedr69** [comment]
yes but does it make the quake live teleporter sound?

**Comment by ghost** [comment]
> yes but does it make the quake live teleporter sound?

Send the sound and it might.

**Comment by Jack7D1** [comment]
How does this interact with multiple people on the same tile?
It's a possible scenario that you have many people teleporting on a safe beacon without moving or being moved, which is then emagged and another person is teleported on and frags.

File: code/datums/helper_datums/teleport.dm [hunk 1]
```diff
@@ -246,3 +246,24 @@
...
 	return TRUE
+
+/datum/teleport/instant/science/doTeleport()
+	if(..())
+		if(isliving(teleatom)) // Limit telefrags to living mobs only
+			var/mob/living/ourself = teleatom
+			var/turf/outturf = get_turf(ourself)
+			var/obj/item/beacon/I = locate() in outturf
+			if(I && I.emagged) // If safety checks are off on beacons.
+				for(var/mob/living/L in outturf)
```
Comment (Jack7D1) [inline code/datums/helper_datums/teleport.dm:257]: Nevermind disregard above

**Comment by pedr69** [comment]
>shithub doesn't support ogg

https://files.catbox.moe/g44h2v.ogg

**Comment by WindowsErrors** [comment]
this reminds me of xenonauts when i was organizing the storming of the second deck of the ship so i got the MAGSTORMers ready on the elevator pad right when the aliens went down the elevator, telefragged both of them, and then threw bombs into my defensive line while everyone else was fluoride staring because I didn't hire anyone with reflexes

i lost 7 people in a single turn

**Pull request merged**